opengrok-indexer/src/main/java/org/opengrok/indexer/Metrics.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opengrok.indexer;
+
+import io.micrometer.core.instrument.binder.jvm.ClassLoaderMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+
+public final class Metrics {
```

Comment:
This is currently specific to the webapp so it should probably live in the `org.opengrok.web` package. I'd like to gather indexer statistics in the future however it will probably use different registry - I think statsD is more suitable for indexer since it allows sending the data rather than waiting for it to be scraped.

Reply:
It's also used in `AuthorizationFramework` which is a part of indexer so moving it under `org.opengrok.web` won't work.

Reply:
You're right. The `AuthorizationFramework` is also part of `RuntimeEnvironment` which makes it even harder to disentangle. Let's leave it like that for now. With the addition of other registries it will have to be made more generic.